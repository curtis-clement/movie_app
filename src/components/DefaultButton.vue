<script setup lang="ts">
const props = withDefaults(defineProps<{
  disabled?: boolean;
}>(), {
  disabled: false,
});

const emit = defineEmits(['buttonClick']);
</script>

<template>
  <button
    class="default-button"
    :disabled="props.disabled"
    @click="emit('buttonClick')"
  >
    <div class="button-content">
      <slot name="text"/>
    </div>
  </button>
</template>

<style scoped>
.default-button {
  background-color: #3498db;
  border: none;
  border-radius: 4px;
  color: white;
  cursor: pointer;
  padding: 0.5rem 1rem;
  transition: background-color 0.2s;
  margin-right: 0.5rem;
}

.default-button:hover {
  background-color: #2980b9;
}

.button-content {
  display: flex;
  align-items: center;
  font-size: .8rem;
  gap: 0.5rem;
}

.default-button:disabled {
  opacity: 0.6;
  pointer-events: none;
}
</style>