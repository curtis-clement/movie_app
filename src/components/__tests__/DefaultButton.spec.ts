import { beforeEach, describe, it, expect } from 'vitest';
import { mount, VueWrapper } from '@vue/test-utils';
import DefaultButton from '@/components/DefaultButton.vue';

const mountComponent = (props: {
  disabled?: boolean;
}) => {
  return mount(DefaultButton, {
    props,
  });
};

describe('DefaultButton', () => {
  let wrapper: VueWrapper<InstanceType<typeof DefaultButton>>;

  beforeEach(() => {
    wrapper = mountComponent({
      disabled: false,
    });
  });

  it('renders the component', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
  
  it('passes props correctly', async () => {
    expect(wrapper.props()).toEqual({
      disabled: false,
    });

    wrapper.setProps({
      disabled: true,
    });

    await wrapper.vm.$nextTick();
    expect(wrapper.props()).toEqual({
      disabled: true,
    });
  });

  it('emits button-click event when clicked', async () => {
    wrapper.find('button').trigger('click');
    await wrapper.vm.$nextTick();
    expect(wrapper.emitted('button-click')).toEqual([[]]);
  });

  it('does not emit button-click event when disabled', async () => {
    wrapper.setProps({
      disabled: true,
    });

    await wrapper.vm.$nextTick();
    wrapper.find('button').trigger('click');
    expect(wrapper.emitted('button-click')).toBeUndefined();
  });
});