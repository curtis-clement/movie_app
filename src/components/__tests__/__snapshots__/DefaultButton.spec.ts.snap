// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DefaultButton > renders the component 1`] = `
"<button data-v-598abd29="" class="default-button">
  <div data-v-598abd29="" class="button-content"></div>
</button>"
`;
